File: server/models/slide.js
var restful = require('node-restful');
var mongoose = restful.mongoose;
var config = require('../config');
var basicAuth = require('../basicAuth')(config.app.user, config.app.pass);

var Slide = restful.model('slides', mongoose.Schema({
    title: 'string',
    body: 'string',
    visible: 'boolean',
    type: 'string',
    index: 'string'
}))
.methods(['get', 'put', 'post', 'delete'])
.updateOptions({'new': true});
Slide.before('put', basicAuth);
Slide.before('post', basicAuth);
Slide.before('delete', basicAuth);
Slide.after('get', (req, res, next) => {
    const program = res.locals.bundle;
    if (Array.isArray(program)) {
        res.locals.bundle = res.locals.bundle.sort((a, b) => parseInt(a.index) > parseInt(b.index));
    }

    next();
});

module.exports = Slide;

File: java-backend/src/main/java/no/javazone/switcharoo/dao/model/Slide.java
package no.javazone.switcharoo.dao.model;

public class Slide {

    public final Long id;
    public final String title;
    public final String body;
    public final boolean visible;
    public final String type;
    public final int index;
    public final String name;
    public final String color;

    public Slide(Long id, String title, String body, boolean visible, String type, int index, String name, String color) {
        this.id = id;
        this.title = title;
        this.body = body;
        this.visible = visible;
        this.type = type;
        this.index = index;
        this.name = name;
        this.color = color;
    }

    public Slide withId(Long id) {
        return new Slide(id, title, body, visible, type, index, name, color);
    }
}
